src/icp_prototype_backend/src/lib.rs

```diff
@@ -120,8 +122,13 @@ impl InterCanisterCallManagerTrait for InterCanisterCallManager {
     async fn icrc1_transfer(
         ledger_principal: Principal,
         req: Icrc1TransferRequest,
-    ) -> CallResult<(Icrc1TransferResponse,)> {
-        ic_cdk::call(ledger_principal, "icrc1_transfer", (req,)).await
```

Comment:
we want to use "transfer" method instead of "icrc1_transfer" as we would not be able to reconstruct `to` value as we need `Principal` value using earlier transaction original sender, which only provides AccountIdentifier in 32 bytes format (no principal). for this purpose we would be using `ic_ledger_types::transfer` instead of `ic_cdk::call`